application/src/main/java/net/corda/v5/application/marshalling/json/JsonSerializer.java

```diff
@@ -1,15 +1,19 @@
-package net.corda.v5.application.marshalling.json
+package net.corda.v5.application.marshalling.json;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
 
 /**
  * An interface to a custom serializer of objects of the specified type T into Json.
  */
-interface JsonSerializer<T> {
+public interface JsonSerializer<T> {
     /**
      * Method called when an object of type T should be serialized.
      *
      * @param item The object to serialize.
      * @param jsonWriter An interface to a writer of Json which should be used to translate fields in item of type T
      * to Json.
      */
-    fun serialize(item: T, jsonWriter: JsonWriter)
+    void serialize(T item, @NotNull JsonWriter jsonWriter) throws IOException;
```

Comment:
Are we sure we want to add `IOException` here? Especially since its a checked exception.

Reply:
We were throwing `IOException` anyway without realising it (assuming the Javadocs were telling the truth about `JsonWriter`'s methods).

Reply:
I've removed the `IOException` :+1:.